Jnario Maven Plugin cannot build specs with Richstrings

Apparently something regarding the org.jnario.richstring-plugins is missing for the maven build. Any spec using a richstring will fail to be built from maven:

   `[ERROR] Failed to execute goal org.jnario:jnario-maven-plugin:1.1.1-SNAPSHOT:testCompile (set-output-directory) on project org.jnario.maven.example: Execution set-output-directory of goal org.jnario:jnario-maven-plugin:1.1.1-SNAPSHOT:testCompile failed: java.lang.IllegalStateException: Unresolved proxy http://www.eclipse.org/xbase/RichString#//RichString. Make sure the EPackage has been registered. -> [Help 1]`

--- a/examples/org.jnario.maven.example/src/test/java/demo/Calculator.spec
+++ b/examples/org.jnario.maven.example/src/test/java/demo/Calculator.spec
@@ -23,4 +23,12 @@ describe "Calculator"{
 		examples.forEach[a + b => sum]
 	}	
 
+	fact "load a richstring" {
+		examples.forEach[
+			val x = '''pre «b» post'''
+			x => "pre " + b + " post"
+		]
+
+	}
+
 }

--- a/plugins/jnario-maven-plugin/src/main/java/org/jnario/maven/JnarioMavenStandaloneSetup.java
+++ b/plugins/jnario-maven-plugin/src/main/java/org/jnario/maven/JnarioMavenStandaloneSetup.java
@@ -3,13 +3,13 @@ import java.util.ArrayList;
 import java.util.List;
 
 import org.eclipse.emf.ecore.EPackage;
-import org.eclipse.xtext.xbase.annotations.XbaseWithAnnotationsStandaloneSetup;
 import org.jnario.feature.FeatureStandaloneSetupGenerated;
 import org.jnario.feature.feature.FeaturePackage;
 import org.jnario.spec.SpecStandaloneSetupGenerated;
 import org.jnario.spec.spec.SpecPackage;
 import org.jnario.suite.SuiteStandaloneSetupGenerated;
 import org.jnario.suite.suite.SuitePackage;
+import org.jnario.xbase.richstring.XbaseWithRichstringStandaloneSetup;
 
 import com.google.inject.Guice;
 import com.google.inject.Injector;
@@ -23,7 +23,7 @@ public class JnarioMavenStandaloneSetup {
 			return injectors;
 		
 		List<Injector> injectors = new ArrayList<Injector>();
-		XbaseWithAnnotationsStandaloneSetup.doSetup();
+		XbaseWithRichstringStandaloneSetup.doSetup();
 		
 		injectors.add(registerFeature());
 		injectors.add(registerSpec());

--- a/plugins/org.jnario.feature/src/org/jnario/feature/FeatureStandaloneSetup.java
+++ b/plugins/org.jnario.feature/src/org/jnario/feature/FeatureStandaloneSetup.java
@@ -12,6 +12,7 @@ import org.eclipse.emf.ecore.EPackage;
 import org.eclipse.xtext.xbase.annotations.xAnnotations.XAnnotationsPackage;
 import org.jnario.JnarioPackage;
 import org.jnario.feature.feature.FeaturePackage;
+import org.jnario.xbase.richstring.xbasewithrichstring.XbaseWithRichstringPackage;
 
 import com.google.inject.Injector;
 
@@ -27,6 +28,7 @@ public class FeatureStandaloneSetup extends FeatureStandaloneSetupGenerated{
 	@Override
 	public Injector createInjectorAndDoEMFRegistration() {
 		EPackage.Registry.INSTANCE.put(XAnnotationsPackage.eINSTANCE.getNsURI(),XAnnotationsPackage.eINSTANCE);
+		EPackage.Registry.INSTANCE.put(XbaseWithRichstringPackage.eNS_URI, XbaseWithRichstringPackage.eINSTANCE);
 		EPackage.Registry.INSTANCE.put(JnarioPackage.eNS_URI, JnarioPackage.eINSTANCE);
 		EPackage.Registry.INSTANCE.put(FeaturePackage.eNS_URI, FeaturePackage.eINSTANCE);
 		return super.createInjectorAndDoEMFRegistration();

--- a/plugins/org.jnario.spec/src/org/jnario/spec/SpecStandaloneSetup.java
+++ b/plugins/org.jnario.spec/src/org/jnario/spec/SpecStandaloneSetup.java
@@ -11,6 +11,7 @@ package org.jnario.spec;
 import org.eclipse.emf.ecore.EPackage;
 import org.jnario.JnarioPackage;
 import org.jnario.spec.spec.SpecPackage;
+import org.jnario.xbase.richstring.xbasewithrichstring.XbaseWithRichstringPackage;
 
 import com.google.inject.Injector;
 
@@ -26,6 +27,7 @@ public class SpecStandaloneSetup extends SpecStandaloneSetupGenerated{
 
 	@Override
 	public Injector createInjectorAndDoEMFRegistration() {
+		EPackage.Registry.INSTANCE.put(XbaseWithRichstringPackage.eNS_URI, XbaseWithRichstringPackage.eINSTANCE);
 		EPackage.Registry.INSTANCE.put(JnarioPackage.eNS_URI, JnarioPackage.eINSTANCE);
 		EPackage.Registry.INSTANCE.put(SpecPackage.eNS_URI, SpecPackage.eINSTANCE);
 		return super.createInjectorAndDoEMFRegistration();

--- a/plugins/org.jnario.xbase.richstring/src/org/jnario/xbase/richstring/XbaseWithRichstringStandaloneSetup.java
+++ b/plugins/org.jnario.xbase.richstring/src/org/jnario/xbase/richstring/XbaseWithRichstringStandaloneSetup.java
@@ -3,6 +3,11 @@
  */
 package org.jnario.xbase.richstring;
 
+import org.eclipse.emf.ecore.EPackage;
+import org.jnario.xbase.richstring.xbasewithrichstring.XbaseWithRichstringPackage;
+
+import com.google.inject.Injector;
+
 /**
  * Initialization support for running Xtext languages 
  * without equinox extension registry
@@ -11,6 +16,13 @@ public class XbaseWithRichstringStandaloneSetup extends XbaseWithRichstringStand
 
 	public static void doSetup() {
 		new XbaseWithRichstringStandaloneSetup().createInjectorAndDoEMFRegistration();
+	}
+	
+	
+	@Override
+	public Injector createInjectorAndDoEMFRegistration() {
+		EPackage.Registry.INSTANCE.put(XbaseWithRichstringPackage.eNS_URI, XbaseWithRichstringPackage.eINSTANCE);
+		return super.createInjectorAndDoEMFRegistration();
 	}
 }
 

